### PR TITLE
Add support for HCL and JSON configs

### PR DESCRIPTION
**Pull request by ghost**
This change required a couple of ugly changes:

* The HCL had to be flattened since every value was ending up in a list
* The HCL parser did not support OrderedDict so a class from the parser was copied and modified

Flattening every list with one element seems likely to break things at some point, so I'm not sure it's worth it but it will only affect HCL files.

DA-1575

### CODE REVIEW

File: tfworker/commands/root.py [hunk 1]
```diff
@@ -163,6 +168,32 @@ def get_config_var_dict(config_vars):
...
+        for i in range(len(hcl)):
+            hcl[i] = replace_hcl_vars(hcl[i], args)
+    elif isinstance(hcl, str):
+        return replace_vars(hcl, args)
+    return hcl
+
+
+def ordered_config_load_hcl(stream, args):
+    """
+    Load a yaml config, and replace templated items.
```
Comment (jessewiles) [inline tfworker/commands/root.py]: Load a ~yaml~ hcl config, ??

File: tfworker/commands/root.py [hunk 2]
```diff
@@ -163,6 +168,32 @@ def get_config_var_dict(config_vars):
...
+    Derived from:
+    https://stackoverflow.com/questions/5121931/in-python-how-can-you-load-yaml-mappings-as-ordereddicts
+    """
+    hcl2_ordered = Lark_StandAlone(transformer=hcl2_transformer.OrderedDictTransformer())
+    doc = hcl2_ordered.parse(stream + "\n")
+    return replace_hcl_vars(doc, args)
+
+
 def ordered_config_load(
     stream, args, Loader=yaml.SafeLoader, object_pairs_hook=OrderedDict
```
Comment (jessewiles) [inline tfworker/commands/root.py:198]: does this also works for json?

**Review by jessewiles — COMMENTED** [review]
This looks good to me.  I'm not seeing any parser changes for JSON.  Can you confirm that the `ordered_config_load` works the same way with JSON?

File: tfworker/commands/root.py (hunk 2 above)
Comment (ghost) [reply tfworker/commands/root.py:198]: YAML is a superset of JSON, so it should just work.  The only issue I see is that pyyaml doesn't support YAML 1.2 which fixed things like `no` being `false` that JSON doesn't do.  I haven't had it fail yet, but I'll do some specific testing of the JSON loading.

Comment (ghost) [reply tfworker/commands/root.py:198]: I did some testing and it seems to behave exactly as the same as when it's loading YAML.

**Review by jessewiles — APPROVED** [review]

**Pull request merged**